### PR TITLE
VM: Ignore container config keys

### PR DESCRIPTION
**Pull request by roosterfish**
When applying configuration to a VM, container specific configuration keys should be ignored.

When config settings are applied indirectly through a profile, running VMs should not raise error messages and instead silently ignore the configuration setting.

Fixes https://github.com/lxc/lxd/issues/11888

### CODE REVIEW

**Review by monstermunchkin — APPROVED** [review]

File: lxd/instance/drivers/driver_qemu.go [hunk 1]
```diff
@@ -5129,36 +5139,14 @@ func (d *qemu) Update(args db.InstanceArgs, userRequested bool) error {
...
-			}
-
-			if strings.HasPrefix(key, "volatile.") {
+			if shared.StringInSlice(key, liveUpdateKeys) {
 				return true
 			}
 
-			if shared.StringInSlice(key, liveUpdateKeys) {
-				return true
+			for _, prefix := range liveUpdateKeyPrefixes {
```
Comment (tomponline) [inline lxd/instance/drivers/driver_qemu.go:5146]: You can use `shared.StringHasPrefix` here.

Comment (roosterfish) [reply lxd/instance/drivers/driver_qemu.go]: Thanks, much cleaner :) Changed.

**Review by tomponline — APPROVED** [review]
Thanks!

**Comment by roosterfish** [comment]
I have just force pushed to rerun the pipeline since there was a single failing job.

**Pull request merged**